src/orbit.jl

```diff
@@ -557,3 +569,164 @@ function Base.show(io::IO, orbit::Orbit)
+
+        file["tau_p"] = t_p
+        file["tau_t"] = t_t
+
+        file["class"] = classes
+
+        file["m"] = orbs[1].coordinate.m
+        file["q"] = orbs[1].coordinate.q
+
+        file["vacuum"] = orbs[1].path.vacuum
+        file["drift"] = orbs[1].path.drift
+
+        if write_paths
+            for i = 1:length(orbs)
+                file[string("path_e",i)] = path_e[i]
+                file[string("path_p",i)] = path_p[i]
+                file[string("path_r",i)] = path_r[i]
+                file[string("path_z",i)] = path_z[i]
+                file[string("path_phi",i)] = path_phi[i]
+                file[string("path_dt",i)] = path_dt[i]
+                file[string("jacdets",i)] = jacdets[i]
+            end
+        end
+
+    end
+    nothing
+end
+
+function read_Orbs(filename::String; old::Bool=false, read_paths::Bool=true)
+    (old||(last(filename,2)=="h5")) && (return read_OrbsOld(filename,read_paths=read_paths))
+
+    isfile(filename) || error("File does not exist")
+    @load filename orbs
+
+    return orbs
+end
+
+function read_OrbsOld(filename; read_paths = true)
+    isfile(filename) || error("File does not exist")
+
+    f = h5open(filename)
+    Em = read(f["Em"])
+    pm = read(f["pm"])
+    Rm = read(f["Rm"])
+    Zm = read(f["Zm"])
+    tm = read(f["tm_normalised"])
+
+    t_p = read(f["tau_p"])
+    t_t = read(f["tau_t"])
+
+    classes = Symbol.(read(f["class"]))
+
+    m = read(f["m"])
+    q = read(f["q"])
+
+    vacuum = read(f["vacuum"])
+    drift = read(f["drift"])
+
+    if read_paths
+        path_e = Vector{Float64}[]
+        path_p = Vector{Float64}[]
+        path_r = Vector{Float64}[]
+        path_z = Vector{Float64}[]
+        path_phi = Vector{Float64}[]
+        path_dt = Vector{Float64}[]
+        jacdets = Vector{Float64}[]
+
+        for i=1:length(Em)
+            push!(path_e,read(f[string("path_e",i)]))
+            push!(path_p,read(f[string("path_p",i)]))
+            push!(path_r,read(f[string("path_r",i)]))
+            push!(path_z,read(f[string("path_z",i)]))
+            push!(path_phi,read(f[string("path_phi",i)]))
+            push!(path_dt,read(f[string("path_dt",i)]))
+            push!(jacdets,read(f[string("jacdets",i)]))
+        end
+    end
+
+    close(f)
+
+    orbs = Orbit{Float64, EPRCoordinate{Float64}}[]
+
+    print("Appending Orbits\n")
+
+    @showprogress for i=1:length(Em)
+        c = EPRCoordinate(Em[i],pm[i],Rm[i],Zm[i],tm[i],m,q)
+
+        if read_paths
+            path = OrbitPath(vacuum,drift,path_e[i],path_p[i],path_r[i],path_z[i],path_phi[i],path_dt[i],jacdets[i])
+            o_i = Orbit(c,classes[i],t_p[i],t_t[i],path,true)
+        else 
+            nullPath = OrbitPath(;vacuum=vacuum,drift=drift)
+            o_i = Orbit(c,classes[i],t_p[i],t_t[i],nullPath,true)
+        end
+
+        push!(orbs,o_i)
+    end
+
+    filename_new = string(chop(filename,tail=2),"jld2")
+    write_Orbs(orbs; filename=filename_new)
```

Comment:
Wait, so when you are done loading orbits from an .h5 file, you automatically re-write them into a .jld2 file?

Reply:
This was to update my hdf5 files, note write_OrbsOLD isn't exported in OrbitTomography.jl
Using jld2 speeds things up massively because I don't have to convert orbits into a matrix of floats

Reply:
*sorry I mean read_OrbsOld isn't exported by GuidingCenterOrbits.jl

Reply:
Ok, that makes sense